cmd/kubelet/app/options/options.go

```diff
@@ -394,6 +391,10 @@ func (f *KubeletFlags) AddFlags(mainfs *pflag.FlagSet) {
 	fs.MarkDeprecated("chaos-chance", "will be removed in a future version.")
 	fs.StringVar(&f.SeccompProfileRoot, "seccomp-profile-root", f.SeccompProfileRoot, "<Warning: Alpha feature> Directory path for seccomp profiles.")
 	fs.MarkDeprecated("seccomp-profile-root", "will be removed in 1.23, in favor of using the `<root-dir>/seccomp` directory")
+	fs.StringVar(&f.CloudProvider, "cloud-provider", f.CloudProvider, "The provider for cloud services. Specify empty string for running with no cloud provider. If set, the cloud provider determines the name of the node (consult cloud provider documentation to determine if and how the hostname is used).")
```

Comment:
Quibble. You can also set the cloud provider "external" if you don't want the Kubelet to have a cloud provider. Setting it to "" will give you a few warnings saying the cloud provider was not set, in case it was something like a typo on the flag name. Setting it to "external" will skip the warnings as we assume its what you wanted at that point.

Reply:
@knabben, let's update the help text to recommend setting to "external" for no cloud provider.